amaranth/build/plat.py

```diff
@@ -328,7 +328,10 @@ def options(opts):
                 return " ".join(opts)
 
         def hierarchy(signal, separator):
-            return separator.join(self._name_map[signal][1:])
+            if isinstance(signal, IOPort):
```

Comment:
Perhaps this is better?

```suggestion
            if isinstance(net, IOPort):
```